### PR TITLE
Created skeletons for the node, host, and router classes

### PR DESCRIPTION
**Pull request by malavika1**

### CODE REVIEW

File: host.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from node import Node
+
+class Host(Node):
+    """A class that will represent hosts.
+
+    Args:
```
Comment (hatooku) [inline host.py]: No need for args if it's the same as the attributes

File: host.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        ns: An instance of the NetworkSimulator class.
+        node_id: The network address of the host.
+        link: The link that the host is connected to.
+        flows: All the flows that are going out of the host.
+
+    Attributes:
```
Comment (hatooku) [inline host.py]: For attributes, put the type of the attribute afterwards, like:

ns (NetworkSimulator): ...
node_id (int): ...

File: host.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @property
+    def link(self):
+        return self._link
+
+    @link.setter
+    def link(self, value):
+        raise AttributeError("Cannot modify the host's link")
+
+    def __init__(self, ns, node_id, link, flows):
```
Comment (hatooku) [inline host.py]: The constructor should be the first method in the class.

File: host.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        Node.__init__(self, ns, node_id)
+        self._link = link
+        self.flows = flows
+
+    def send_packet(self, packet):
+        """Sends a packet to another node.
+
+        Args:
+            packet: The packet to send.
```
Comment (hatooku) [inline host.py]: Same thing here, ex.

packet (Packet): ...

File: node.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+class Node(object):
+    """A node class that will be the parent of the hosts and routers.
+
+    Args:
```
Comment (hatooku) [inline node.py]: No args needed

File: node.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        ns: An instance of the NetworkSimulator class.
+        node_id: The network address of the host or router.
+
+    Attributes:
```
Comment (hatooku) [inline node.py]: Attribute type after name, e.g. ns (NetworkSimulator)

File: node.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    @property
+    def node_id(self):
+        return self._node_id
+
+    @node_id.setter
+    def node_id(self, value):
+        raise AttributeError("Cannot modify network address")
+
+    def __init__(self, ns, node_id):
```
Comment (hatooku) [inline node.py]: __init__ should be first method

File: router.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+from node import Node
+
+class Router(Node):
+    """A class that will represent routers.
+
+    Args:
```
Comment (hatooku) [inline router.py]: No need for args

File: router.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    def send_packet(self, packet):
+        """Sends a packet to another node.
+
+        Args:
+            packet: The packet to send.
+
+        """
+        pass
+
+    def receive_packet(self, packet):
```
Comment (hatooku) [inline router.py]: Specify that receive_packet will call send_packet

File: host.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+        """
+        pass
+
+    def receive_packet(self, packet):
+        """Receives a packet from another node.
```
Comment (mortonrj) [inline host.py]: Say is the methods return a thing if they return a thing?

Comment (malavika1) [inline host.py]: idk if it returns a thing :( does it????

File: node.py [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        self.ns = ns
+        self._node_id = node_id
+
+    def send_packet(self, packet):
```
Comment (Jdonerme) [inline node.py]: Does python not have a way to make stuff abstract?

File: host.py (hunk 10 above)
Comment (mortonrj) [inline host.py]: Oh idk

**Comment by malavika1** [comment]
Addressed changes

**Pull request merged**